content/academy/web_scraping_for_beginners/best_practices.md

```diff
@@ -0,0 +1,126 @@
+
+> To learn more about some of the most popular (and awesome) ES6+ features, check out [this](https://medium.com/@matthiasvstephens/why-is-es6-so-awesome-88bff6857849) article.
+
+### [](#no-magic-numbers) No magic numbers
+
+Avoid using [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) as much as possible. Either declare them as a **constant** variable in your **constants.js** file, or if they are only used once, add a comment explaining what the number is.
+
+Don't write code lke this:
+
+```JavaScript
+const x = (y) => (y - 32) * (5 / 9);
+```
+
+That is quite confusing due to the undescriptive naming and the magic numbers. Do this instead:
+
+```JavaScript
+// Converts a fahrenheit value to celsius
+const fahrenheitToCelsius = (celsius) => (celsius - 32) * (5 / 9);
+```
+
+> Additionally, try to make values for options such as `maxRequestRetries` in the Apify SDK configurable via input. If they absolutely must be a strict value, make them a constant variable.
+
+### [](#use-comments) Use comments!
+
+Don't be shy to add comments to your code! Even when using descriptive function and variable naming, it might still be a good idea to add a comment in places where you had to make a tough decision or chose an unusual choice.
+
+> If you're a true pro, use [JSDoc](https://jsdoc.app/) to comment and document your code.
+
+## [](#logging) Logging
+
+Logging helps you understand exact what your scraper is doing. Generally, having more logs is better than having less. Especially make sure to log your `catch` blocks - no error should pass unseen unless there is a good reason.
+
+For scrapers than will run longer than usual, keep track of some useful stats (such as **itemsScraped** or **errorsHit**) and log them to the console on an interval.
+
+The meaning of your log messages should make sense to an outsider who is not familiar with the inner workings of your scraper. Avoid log lines with just numbers or just URLs - always identify what the number/string means.
+
+Here is an example of an "incorrect" log message:
+
+```text
+300  https://example.com/1234  1234
+```
+
+And here is  that log message translated into something that makes much more sense to the end user:
+
+```text
+Index 1234 --- https://example.com/1234 --- took 300 ms
+```
+
+## [](#input) Input
+
+There are two main best practices when it comes to accepting input into a scraper.
+
+### [](#set-limits) Set limits
+
+When allowing your users to pass input properties which could break the scraper (such as **timeout** set to **0**), be sure to disallow ridiculous values. Set a maximum/minimum number allowed, maximum array input length, etc.
+
+### [](#validate) Validate
+
+Validate the input provided by the user! This should be the very first thing your scraper does. If the fields in the input are missing or in an incorrect type/format, either parse the value and correct it programmatically or throw an informative error telling the user how to fix the error.
```

Comment:
I would add a callout that says that on the apify platform you can use input schema (with a link) to validate inputs and also generate a nice UI